client_ext_test.go

```diff
@@ -227,6 +227,58 @@ func TestGetNoContentHeaders(t *testing.T) {
 	assert.Equal(t, http.MethodGet, unaryReq.HTTPMethod())
 }
 
+func TestConnectionDropped(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	for _, protocol := range []string{connect.ProtocolConnect, connect.ProtocolGRPC, connect.ProtocolGRPCWeb} {
```

Comment:
The protocol doesn't affect the test case, maybe that too implementation aware but this looks like it can be flattened to the one testcase.

Reply:
I don't understand this remark. What one test case would I flatten it to?

It does, in fact, effect the test case: we use this to build up client options, to select the protocol, and then pass those options below (line 249).

Reply:
Setting the protocols doesn't seem to affect what the test does, but this is an implementation detail. So this testcase might be better suited as an internal test on duplexHTTPCall that `makeRequest` returns the correct error response.

Reply:
> doesn't seem to affect what the test does

I still don't understand what you mean. They cause the client to change the protocol it uses, to make sure that the error is handled correctly through all protocol implementations, for both unary and streaming RPCs.

I definitely don't want it to just be a test of `makeRequest`, because somewhere further up the stack could be changed in a way that breaks this expectation. We want to assert that the "unavailable" error makes it all the way up to the RPC calling code. Previously, the "unavailable" error was getting _changed_ to "unimplemented" (and that happens _outside_ of `makeRequest`).

Reply:
Okay, this makes sense. Better as is.